### ISSUE TITLE
Resolvers with direct_lambda can't be used alongside regular ones

### ISSUE THREAD

**Issue by half2me** [terraform-aws-modules/terraform-aws-appsync#16]
## Description
If one of the resolvers is using `direct_lambda` option, it won't work alongside regular ones.
For example: (1 direct lambda, one normal)
```terraform
resolvers = {
    "Query.foo1" = {
      data_source   = "lambda"
      direct_lambda = true
    }

    "Query.foo2" = {
      data_source       = "lambda"
      request_template  = file("foo.req.vtl")
      response_template = file("foo.res.vtl")
    }
  }
}
```

It gives me this error:
```
Error: Inconsistent conditional result types
│ 
│   on .terraform/modules/appsync/main.tf line 2, in locals:
│    2:   resolvers = var.create_graphql_api ? { for k, v in var.resolvers : k => merge(v, {
│    3:     type  = split(".", k)[0]
│    4:     field = join(".", slice(split(".", k), 1, length(split(".", k))))
│    5:   }) } : tomap({})
│     ├────────────────
│     │ var.resolvers is object with 2 attributes
│ 
│ The true result value has the wrong type: element types must all match for conversion to map.
```

If all resolvers use `direct_lambda` or none of them use it, it works fine.

## Versions

 - Terraform:
```
Terraform v0.15.4
on darwin_amd64
+ provider registry.terraform.io/hashicorp/aws v3.39.0
+ provider registry.terraform.io/hashicorp/external v2.1.0
+ provider registry.terraform.io/hashicorp/local v2.1.0
+ provider registry.terraform.io/hashicorp/null v3.1.0
+ provider registry.terraform.io/hashicorp/random v3.1.0
```
 - Provider(s):

 - Module:

## Reproduction

No, just local state

Yes

**Comment by nelg** [comment]
also impacted, in my case, with dynamodb resolvers + lambda resolvers.

**Comment by cmpcdaly** [comment]
I'm also impacted, any updates on this?

**Comment by half2me** [comment]
@antonbabenko Any updates?

**Comment by antonbabenko** [comment]
This issue has been resolved in version 1.1.2 :tada:
